lib/text/vtt_text_parser.js

```diff
@@ -219,6 +221,46 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {!shaka.text.Cue} cue
+   * @param {string} payload
+   */
+  static parseCueStyles(cue, payload) {
+    try {
+      const parser = new DOMParser();
+      const xml = parser.parseFromString(payload, 'text/xml');
+      const parserError = xml.getElementsByTagName('parsererror')[0];
+      if (parserError) {
+        throw new shaka.util.Error(
+            shaka.util.Error.Severity.CRITICAL,
+            shaka.util.Error.Category.TEXT,
+            shaka.util.Error.Code.INVALID_XML,
+            parserError.textContent);
+      }
+      const element = xml.documentElement;
+      // TODO: Add support for nested elements at more than two levels of nesting
+      const payloadString = element.textContent.trim();
+      if (payloadString) {
+        cue.payload = payloadString;
+        switch (element.tagName) {
+          case 'b':
```

Comment:
Aren't there other tags like `<b.foo>`?  I don't remember what those mean, but you may want to add a TODO for these.  There are also timestamp tags that we may want to just ignore.

Reply:
There are class tags like `<c.foo>` that are meant to have class names attached to the elements for styling in CSS, and voice tags like `<v Bob>` to show who is talking.  With native cues, both can be used to with a `::cue()` selector to style the cues.

You can also have styles embedded into VTT, which we don't currently support.  Extracting those would be a bit of a pain, and we'd have to inject them into programmatic stylesheets.

https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API#Styling_WebVTT_cues

This is a great start, though, and probably enough for most folks.  We also don't yet have a way to represent classes and voices in `shaka.text.Cue`.

Reply:
Yes, `b.foo` is allowed by the spec, I will change it to allow it

Reply:
At the moment it will only support: b, u, i. But I think that with what I'm preparing it will be easy to expand this list it in the future.